internal/handlers/pg/pgdb/collections.go

```diff
@@ -108,9 +109,14 @@ func CollectionExists(ctx context.Context, tx pgx.Tx, db, collection string) (bo
 // It returns possibly wrapped error:
 //   - ErrInvalidDatabaseName - if the given database name doesn't conform to restrictions.
 //   - ErrInvalidCollectionName - if the given collection name doesn't conform to restrictions.
+//   - ErrCollectionStartsWithDot - if the given collection name starts with dot.
 //   - ErrAlreadyExist - if a FerretDB collection with the given name already exists.
 //   - *transactionConflictError - if a PostgreSQL conflict occurs (the caller could retry the transaction).
 func CreateCollection(ctx context.Context, tx pgx.Tx, db, collection string) error {
+	if strings.HasPrefix(collection, ".") {
+		return ErrCollectionStartsWithDot
+	}
+
```

Comment:
I wanted to use regex for this check, although MongoDB returns different error for this case.